openshift/machine-config-operator
# OCPBUGS-38869: MCN are not updated when we use OCL

**- What I did**
Fixed a bug in `updateOnClusterBuild` where mcn's were not correctly updating their `.spec.configVersion.desired` field when using ocl. The fix involved leveraging the GenerateAndApplyMachineConfigNodeSpec function to ensure that the MCNs are correctly updated with the latest desired configuration after applying a new MachineConfig.

**- How to verify it**
1. Enable ocl
2. Apply a new mc that adds or modifies a file on the worker nodes.
3. Verify that the `.spec.configVersion.desired` field of mcn's reflects the updated mc across all nodes in the worker pool.

**- Description for the changelog**
Fixed an issue where mcn's were not updating their `.spec.configVersion.desired` field correctly when using ocl in the mcd
